### PR TITLE
*: support get property

### PR DESCRIPTION
**Pull request by siddontang**
@ngaut @BusyJay

### CODE REVIEW

File: src/rocksdb.rs [hunk 1]
```diff
@@ -814,6 +814,65 @@ impl DB {
...
+                                                           prop_name.as_ptr())
+                }
+            };
+
+            if value.is_null() {
+                return None;
+            }
+
+            // Must valid UTF-8 format.
+            let s = CStr::from_ptr(value).to_str().unwrap().to_owned();
```
Comment (BusyJay) [inline src/rocksdb.rs]: Can the copy be avoided?

File: src/rocksdb.rs [hunk 2]
```diff
@@ -814,6 +814,65 @@ impl DB {
...
+            libc::free(value as *mut c_void);
+            Some(s)
+        }
+    }
+
+    fn get_property_int_cf_opt(&self,
+                               cf: Option<DBCFHandle>,
+                               name: &str)
+                               -> Option<u64> {
+        // Rocksdb guarantee that the return property int
```
Comment (BusyJay) [inline src/rocksdb.rs]: Why and how? What if I want to set a negative number?

Comment (siddontang) [inline src/rocksdb.rs]: Rocksdb returns uint64 for all int property value, no negative.

File: src/rocksdb.rs (hunk 1 above)
Comment (siddontang) [inline src/rocksdb.rs]: I think it is no need here, the property value is very small, it can't be the performance bottleneck.

**Comment by BusyJay** [comment]
LGTM

**Comment by siddontang** [comment]
@diskings

**Comment by siddontang** [comment]
@disksing @tiancaiamao

File: librocksdb_sys/src/lib.rs [hunk 3]
```diff
@@ -517,6 +529,21 @@ mod test {
...
+
+            let sst_size =
+                CStr::from_ptr(value).to_str().unwrap().parse::<u64>().unwrap();
+            assert!(sst_size > 0);
+            libc::free(value as *mut c_void);
+
+            let propname = CString::new("fake_key").unwrap();
+            let value = rocksdb_property_value(db, propname.as_ptr());
+            assert!(value.is_null());
+            libc::free(value as *mut c_void);
```
Comment (disksing) [inline librocksdb_sys/src/lib.rs]: Need free for the null value?

Comment (siddontang) [inline librocksdb_sys/src/lib.rs]: no need.

**Comment by disksing** [comment]
LGTM.

**Pull request merged**